lms/resources/lti_launch.py

```diff
@@ -122,8 +122,19 @@ def h_group_name(self):
         :raise HTTPBadRequest: if an LTI param needed for generating the group
           name is missing
         """
-        name = self._get_param("context_title").strip()
+        return self._group_name(self._get_param("context_title").strip())
 
+    def h_section_group_name(self, section):
+        """
+        Return the h group name for the given Canvas course section.
+
+        :param section: a section dict as received from the Canvas API
+        """
+        return self._group_name(section["name"].strip())
+
+    @staticmethod
+    def _group_name(name):
```

Comment:
The separate `h_group_name`, `h_section_group_name`, `h_groupid` and `h_section_groupid` methods are all going to be merging together into a single `h_groups` method soon (and that method will have to take a sections-or-courses option or make that decision itself).

For now, this PR just puts in place the method of generating section group names: by taking the section's name and truncating it.